examples/ssd/train.py

```diff
@@ -0,0 +1,159 @@
+
+    def transform(in_data):
+        img, bbox, label = in_data
+
+        img = random_distort(img)
+
+        if np.random.randint(2):
+            img, param = transforms.random_expand(
+                img, fill=mean, return_param=True)
+            bbox = transforms.translate_bbox(
+                bbox, y_offset=param['y_offset'], x_offset=param['x_offset'])
+
+        img, param = random_crop_with_bbox_constraints(
+            img, bbox, return_param=True)
+        bbox, param = transforms.crop_bbox(
+            bbox, y_slice=param['y_slice'], x_slice=param['x_slice'],
+            allow_outside_center=False, return_param=True)
+        label = label[param['index']]
+
+        _, H, W = img.shape
+        img = resize_with_random_interpolation(img, (size, size))
+        bbox = transforms.resize_bbox(bbox, (H, W), (size, size))
+
+        img, params = transforms.random_flip(
+            img, x_random=True, return_param=True)
+        bbox = transforms.flip_bbox(
+            bbox, (size, size), x_flip=params['x_flip'])
+
+        img -= np.array(mean)[:, np.newaxis, np.newaxis]
+        mb_loc, mb_label = coder.encode(
+            transforms.resize_bbox(bbox, (size, size), (1, 1)), label)
+        return img, mb_loc, mb_label
+
+    train = TransformDataset(
+        ConcatenatedDataset(
+            VOCDetectionDataset(year='2007', split='trainval'),
+            VOCDetectionDataset(year='2012', split='trainval')
+        ),
+        transform)
+    train_iter = chainer.iterators.MultiprocessIterator(
+        train, args.batchsize, n_processes=2)
+
+    test = VOCDetectionDataset(
+        year='2007', split='test',
+        use_difficult=True, return_difficult=True)
+    test_iter = chainer.iterators.SerialIterator(
+        test, args.batchsize, repeat=False, shuffle=False)
+
+    optimizer = chainer.optimizers.MomentumSGD()
+    optimizer.setup(train_chain)
+    for param in train_chain.params():
+        if param.name == 'b':
+            param.update_rule.add_hook(GradientScaling(2))
+        else:
+            param.update_rule.add_hook(WeightDecay(0.0005))
+
+    updater = training.StandardUpdater(train_iter, optimizer, device=args.gpu)
+    trainer = training.Trainer(updater, (120000, 'iteration'), args.out)
+    trainer.extend(
+        extensions.ExponentialShift('lr', 0.1, init=1e-3),
+        trigger=triggers.ManualScheduleTrigger([80000, 100000], 'iteration'))
+
+    trainer.extend(
+        DetectionVOCEvaluator(
+            test_iter, model, use_07_metric=True,
+            label_names=voc_detection_label_names),
+        trigger=(10000, 'iteration'))
+
+    log_interval = 10, 'iteration'
+    trainer.extend(extensions.LogReport(trigger=log_interval))
+    trainer.extend(extensions.observe_lr(), trigger=log_interval)
+    trainer.extend(extensions.PrintReport(
+        [
+            'epoch', 'iteration',
+            'main/loss', 'main/loss/loc', 'main/loss/conf',
+            'validation/main/map', 'lr']),
```

Comment:
Oh I see.
From the reasons I mentioned above, we should put `lr` before performance reports.